### PR TITLE
Refactor instrumentdata, add more tests

### PR DESCRIPTION
**Pull request by rerpha**
This PR refactors the instrumentData class to utilise typescript interfaces and use them to store PV data in the `instrument` instance. 

it also adds some unit/UI tests, though we still have a fair way to go to get full coverage. I stopped here because this was going to get bigger and bigger if I was aiming for 100% off the bat. 

note the snapshots are deliberately committed as per https://jestjs.io/docs/snapshot-testing#should-snapshot-files-be-committed

### CODE REVIEW

**Review by Tom-Willemsen — COMMENTED** [review]
Looking a lot nicer! Few questions/comments, but this is definitely an improvement...

<details>
<summary>Do we need to worry about these sorts of warnings? (When running npm run test)</summary>

```
    console.error
      Warning: validateDOMNesting(...): <tr> cannot appear as a child of <div>.
          at tr

      13 |   };
      14 |   const instName = "Instrument";
    > 15 |   const { container } = render(Block(aBlock, instName, false));
         |                               ^
      16 |   expect(container).toMatchSnapshot();
      17 | });
      18 |

      at printWarning (node_modules/react-dom/cjs/react-dom.development.js:86:30)
      at error (node_modules/react-dom/cjs/react-dom.development.js:60:7)
      at validateDOMNesting (node_modules/react-dom/cjs/react-dom.development.js:10847:7)
      at createInstance (node_modules/react-dom/cjs/react-dom.development.js:10930:5)
      at completeWork (node_modules/react-dom/cjs/react-dom.development.js:22226:28)
      at completeUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:26632:16)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:26607:5)
      at workLoopSync (node_modules/react-dom/cjs/react-dom.development.js:26505:5)
      at renderRootSync (node_modules/react-dom/cjs/react-dom.development.js:26473:7)
      at performConcurrentWorkOnRoot (node_modules/react-dom/cjs/react-dom.development.js:25777:74)
      at flushActQueue (node_modules/react/cjs/react.development.js:2667:24)
      at act (node_modules/react/cjs/react.development.js:2582:11)
      at actWithWarning (node_modules/react-dom/cjs/react-dom-test-utils.development.js:1740:10)
      at node_modules/@testing-library/react/dist/act-compat.js:46:25
      at renderRoot (node_modules/@testing-library/react/dist/pure.js:161:26)
      at render (node_modules/@testing-library/react/dist/pure.js:247:10)
      at Object.<anonymous> (app/components/Block.test.ts:15:31)
```

</details>

File: app/components/TopBar.tsx [hunk 1]
```diff
@@ -1,22 +1,33 @@
...
+  return "UNKNOWN";
+}
+
+export default function TopBar({
+  dashboard,
   instName,
   runInfoPVs,
 }: {
-  monitoredPVs: Map<string, any>;
+  dashboard: Array<Array<Array<IfcPV>>>;
```
[the remark this answers is not in the document]
Comment (rerpha) [reply app/components/TopBar.tsx:22]: I think we can use dashboardArr yes - must have missed this out.

File: app/components/TopBar.tsx [hunk 2]
```diff
@@ -34,115 +45,73 @@ const TopBar = ({
...
           <span className="block max-h-14 overflow-hidden rounded-lg bg-gray-50 hover:bg-gray-800 hover:text-white px-4 py-0 mb-2  shadow-lg transition-all duration-300 peer-checked/showLabel:max-h-fit cursor-pointer">
             <h3 className="flex h-14 cursor-pointer items-center font-bold ">
               Click to show/hide all run information
             </h3>
-            {Array.from(runInfoPVs.entries()).map((runInfoPV) => (
-              <p className="mb-2" key={runInfoPV[0]}>
-                {runInfoPV[0]}: {runInfoPV[1]}
+            {runInfoPVs.map((runInfoPV) => (
+              <p className="mb-2" key={runInfoPV.human_readable_name}>
+                {runInfoPV.human_readable_name}: {runInfoPV.value}
```
[the remark this answers is not in the document]
Comment (rerpha) [reply app/components/TopBar.tsx:109]: https://github.com/ISISComputingGroup/WebDashboard/issues/55

**Review by Tom-Willemsen — APPROVED** [review]
Looking like a substantial improvement - as you say could go further with coverage etc but this PR is already big enough.

**Pull request merged**